src/components/tx-flow/index.tsx

```diff
@@ -41,29 +43,40 @@ export const TxModalProvider = ({ children }: { children: ReactNode }): ReactEle
     }
 
     const ok = confirm('Closing this window will discard your current progress.')
-    if (ok) {
-      handleModalClose()
+    if (!ok) {
+      return
     }
+
+    // Reject if the flow is closed
+    txDispatch(TxEvent.USER_QUIT, undefined)
+
+    handleModalClose()
   }, [shouldWarn, handleModalClose])
 
   const setTxFlow = useCallback(
     (txFlow: TxModalContextType['txFlow'], onClose?: () => void, shouldWarn?: boolean) => {
+      // Reject if a flow is open and the user changes to a different one
+      const isSuccesFlow = txFlow?.type === SuccessScreen
+      if (flow && flow !== txFlow && !isSuccesFlow) {
```

Comment:
```suggestion
      if (flow && flow !== txFlow && txFlow?.type === SuccessScreen) {
```
IMHO easier to read and no unnecessary vars.

Reply:
Addressed in 31f019ca951be8f57093fdfda0d388e3a1ba0f65.